source/features/link-to-gihub-io.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "dom-chef";
+import elementReady from "element-ready";
+import select from "select-dom";
+import * as pageDetect from "github-url-detection";
+import LinkExternalIcon from "octicon/link-external.svg";
+
+import features from ".";
+
+function getCurrentUsername(): string {
+	return location.pathname.replace(/(^\/)/g, "").split("/")[0];
+}
+
+function getGithubIoLink(username: string): string {
+	return `${username}.github.io`;
+}
+
+async function repoTabInit(): Promise<void> {
+	const username = getCurrentUsername();
+	const githubIoLink = getGithubIoLink(username);
+	const repo = select(`a.rgh-discussion-links[href$="${githubIoLink}"]`);
```

Comment:
Selecting the exact URL is not strictly necessary, we just need to select any repos that end in `.github.io`. **This also applies to `githubIoRepoInit`**

Also `a.rgh-discussion-links` is added by another feature, we usually don't make features inter-dependent, so it needs to be changed

Reply:
Hi, so I've switched to using this selector
```js
const repos = select.all('a[href$=".github.io"][itemprop="name codeRepository"]');
```
After which I run a forEach to add a `external-link` to each one (incase of multiple github.io's)

However, to know what URL the page needs to be redirected to, I need the name / URL it is pointing to, for which I am doing something like this:
```js
repos.forEach(function(repo) {
		repo!.parentElement!.append(
			<a
				href={repo.href} // This line
				target="_blank"
				rel="noopener noreferrer"
			>
				<LinkExternalIcon/>
			</a>
		);

	})
```
This is currently just to point the error. In the DOM the href points to the code repository (from which we can extract the URL), however when trying to access href from repo, I am thrown with a error from Typescript: `Property 'href' does not exist on type 'HTMLElement'.`.

I am guessing this is something to do with how `select-dom` is built, is there anything I'm missing / any work around?

Reply:
Ignore the typescript errors, we can fix those later.

When running `npm run watch` the extention will build even if there are typescript errors

Reply:
Okay, pushed with this change